### ISSUE TITLE
Upgrade to API v3

### ISSUE THREAD

**Issue by AlexandreRio** [tobold/habirc#1]
Hi,

I just started using your module to link an IRC channel and I noticed that it still use the v2 API that will soon be stopped. Do you plan to move to the v3 ?

Using the api_url to point to the new URL does not work when trying to get information about hero (first query I guess).

**Comment by kindfulkirby** [comment]
Yeah, it's on my plate. I hope to get around to it next week :)

**Comment by AlexandreRio** [comment]
Cool :smile:

Off topic:
- Are the api_user and api_key variables optional? Since we have to use `.hero add`.
- The bot is supposed to read for new messages in the habitica chat and post them to IRC? I only manage to post messages from IRC to habitica with `.say`.

**Comment by kindfulkirby** [comment]
1. No, they are not optional. They are for reading the chat and when users haven't added their API key
2. Have you added api_user and api_key to your config file? Because that's where they are used 😉

Had I known people would actually use this, I'd have put more effort into the documentation 😥

**Comment by AlexandreRio** [comment]
Yes I have them, I also added a `channels` variable with the same value as the one in the [core] section, otherwise (if I remove the channels from the [core]) the bot does not connect at all.

I totally understand that documentation isn't your main focus. I know the feel  :laughing:

**Comment by kindfulkirby** [comment]
Hmm. Can you paste me your config file (minus passwords and API keys, of course)?

**Comment by AlexandreRio** [comment]
```
[core]
nick = botname
host = ircserver
use_ssl = false
port = 6667
owner = mynick
channels = #chan

[habitica]
api_user = user-id
api_key = key-id
channels = #chan
chats = group-ip
```

I tried with the `enable` option but this way I have all the default modules and habitica.

**Comment by kindfulkirby** [comment]
Well, you've **_completely**_ anonymized that, so I can't tell anything for sure.

Does the .hero command work?  
If not, are you sure the bot has voice in your channel?

**Comment by AlexandreRio** [comment]
Sorry about that, what information do you need? The server ones are correct since the bot joined the IRC channel and the habitica also because `.hero` gives me my current health/exp/mana and `.say` post in the right habitica group chat.  I used it with sopel v 6.3.1.

I tried with only habitica enabled and there's no error in the logs.

**Comment by crookedneighbor** [comment]
Just chining in that version 2 of the Habitica API will be officially [deactivated on July 15th](http://devs.habitica.com/api-v2-deprecation-notice/). 

[Here is a migration guide](http://devs.habitica.com/api-v2-v3-migration-guide/) for upgrading to version 3. If you have any questions, feel free to reach out on [Github](https://github.com/HabitRPG/habitrpg/issues) or the [Aspiring Comrades Guild](https://habitica.com/#/options/groups/guilds/2ff9822b-27f2-4774-98da-db349b57a38e).
